### DISCUSSION TITLE
Native renderer?

### DISCUSSION THREAD

**Discussion by shishkin** [tauri-apps/tauri#10964] [Q&A]
I'm new to rust GUIs and looking for a framework to build a cross-platform (iOS, Android, MacOS, Linux, Windows, Web/WASM) app. So far Tauri looks promising. However I was curious if it's possible to use Tauri base with another renderer than HTML inside a WebView. In particular, a native renderer like [iced](https://github.com/iced-rs/iced).

**Comment by Tsuguri** [comment]
I'm not exactly sure if tauri supports WebGL or WebGPU (that could probably just boil down to webview used on your target platform). You can probably pretty easily check that - create tauri app that uses rust for frontend and then put iced/egui web-enabled example into the frontend part. Or even wgpu example (if you can use wgpu then you can render egui/iced if I'm not wrong).

I'm not sure what exactly you want to gain by using something else instead of html. I would say that using web frontend tools is one of the main selling point of tauri (in my opinion).

**Reply by shishkin** [reply]
I'm mostly interested in Tauri as "Flutter in Rust" with its native platform API abstraction, unified bundling and plugin ecosystem. So far, Tauri examples with Rust UI frameworks (Leptos, Sycamore and Yew) all still manipulate DOM from a WASM binary. I'm fine with that when targeting Web, but would prefer less layers of abstraction on native platforms.

**Reply by Tsuguri** [reply]
I'm pretty sure using html manipulation (either js or wasm) is the way to do UI in tauri. It's basically electron with 2 major differences: using system webview and rust backend. I don't think you're going to get "ready out of the box" experience if you try to use it differently. You can probably render everything yourself from backend but then there's no point in using tauri at all.

**Reply by shishkin** [reply]
Yeah, I'm thinking how I can use Tauri building blocks like bundling and platform APIs "à la carte". Rust [GUI ecosystem](https://areweguiyet.com/) seems so fragmented 🤯
